examples/with-apollo/lib/with-apollo.js

```diff
@@ -13,12 +13,23 @@ let apolloClient = null
  * to a next.js PageTree. Use it by wrapping
  * your page component via HOC pattern.
  * @param {Function|Class} PageComponent
+ * @param {Object} [config]
+ * @param {Boolean} [config.ssr=true]
  */
-function withApollo (PageComponent) {
-  return class extends React.Component {
-    static displayName = `withApollo(${getDisplayName(PageComponent)})`
+function withApollo (PageComponent, { ssr = true } = {}) {
+  const WithApollo = ({ apolloClient, apolloState, ...pageProps }) => {
+    const client = useMemo(() => apolloClient || initApollo(apolloState), [])
+    return (
+      <ApolloProvider client={client}>
+        <PageComponent {...pageProps} />
+      </ApolloProvider>
+    )
+  }
+
+  WithApollo.displayName = `withApollo(${getDisplayName(PageComponent)})`
```

Comment:
I think this is causing hot reloading on pages that use WithApollo to reload the page on each change, rather than updating the DOM. I'm trying to find where next js handles the hot-reloading. Presumably the display name has something to do with it. Is it provided by another package?